derive unit.blueIncidentAssigned

heinrichmartin opened the pull request:
As of wrk-fmd/geobroker@6d3e87adbc164c15548094415702d24a6d1c941d, geoclient can derive whether a unit has a blue incident assigned. Funny enough, I don't have a test setup for it.

Please verify in an appropriate environment. See comment on 2a79c99b15e1d7ac4f4f0e2a7b737fad0a7c1df5.

heinrichmartin committed:
handle incidents before units

in preparation of linking units to their incident

heinrichmartin committed:
clarify bad comment describing the algorithm

heinrichmartin committed:
derive unit.blueIncidentAssigned

from incident.assignedUnits and incident.blue

robo-w commented on client/js/public.js:
```diff
@@ -406,8 +406,42 @@
+      // while updating units and incidents, use a Set for the diff with existing ones
+      let toBeRemoved;
+      // incident data containes assigned units, also track the reverse direction
+      let incidents = new Map(); // unit.id => incident
```
Although it is not used very often, it is possible that a unit is assigned to multiple incidents. Is this covered by this map implementation? I think we can get an inconsistent "blue", if one assigned incident is set "blue", and a second one of the same unit not.
I would expect it, that a unit is marked as "blueIncidentAssigned", if at least one of the assigned is blue.

heinrichmartin committed:
allow multiple incidents per unit

The expected fix was a Map from unit.id to an array of incidents.
However, this implementation fulfills #29.
closes #30

heinrichmartin committed:
Merge branch 'develop' into feature/blueIncidentAssigned
